CHANGES.md

```diff
@@ -1,3 +1,150 @@
+  * Added validation for the `ServiceProfile` CRD
+  * Updated the `linkerd check` command to disallow setting both the `--pre` and
+    `--proxy` flags simultaneously
+  * Added `--routes` flag to the `linkerd top` command, for grouping table rows
+    by route instead of by path
+  * Updated Prometheus configuration to automatically load `*_rules.yml` files
+  * Removed TLS column from the `linkerd routes` command output
+  * Updated `linkerd install` output to use non-default service accounts,
+    `emptyDir` volume mounts, and non-root users
+  * Removed cluster-wide resources from single-namespace installs
+  * Fixed resource requests for proxy-injector container in `--ha` installs
+* Controller
+  * Added full stat and tap support for DaemonSets and StatefulSets in the CLI,
+    Grafana, and web UI
+  * Updated auto-injector to use the proxy log level configured at install time
+  * Fixed issue with auto-injector including TLS settings in injected pods even
+    when TLS was not enabled
+  * Changed automatic proxy injection to be opt-in via the `linkerd.io/inject`
+    annotation on the pod or namespace
+  * Move service profile definitions to client and server namespaces, rather
+    than the control plane namespace
+  * Added `linkerd.io/created-by` annotation to the linkerd-cni DaemonSet
+  * Added a 10 second keepalive default to resolve dropped connections in Azure
+    environments
+  * Improved node selection for installing the linkerd-cni DaemonSet
+  * Corrected the expected controller identity when configuring pods with TLS
+  * Modified klog to be verbose when controller log-level is set to `debug`
+  * Added support for retries and timeouts, configured directly in the service
+    profile for each route
+  * Added an experimental CNI plugin to avoid requiring the NET_ADMIN capability
+    when injecting proxies
+  * Improved the API for `ListPods`
+  * Fixed `GetProfiles` API call not returning immediately when no profile
+    exists (resulting in proxies logging warnings)
+  * Blocked controller initialization until caches have synced with kube API
+  * Fixed proxy-api handling of named target ports in service configs
+  * Added parameter to stats API to skip retrieving prometheus stats
+* Web UI
+  * Grayed-out the tap icon for requests from sources that are not meshed
+  * Improved resource detail pages to show all resource types
+  * Fixed stats not appearing for routes that have service profiles installed
+  * Added "meshed" and "no traffic" badges on the resource detail pages
+  * Fixed `linkerd dashboard` to maintain proxy connection when browser open fails
+  * Fixed JavaScript bundling to avoid serving old versions after upgrade
+  * Reduced the size of the webpack JavaScript bundle by nearly 50%
+  * Fixed an indexing error on the top results page
+  * Restored unmeshed resources in the network graph on the resource detail page
+  * Reduced the overall size of the asset bundle for the web frontend
```

Comment:
I feel like this is a repeat of `Reduced the size of the webpack JavaScript bundle by nearly 50%`